tests/Microsoft.Identity.Test.Unit.net45/Integration/DeviceCodeFlowIntegrationTest.cs

```diff
@@ -0,0 +1,146 @@
+using Microsoft.Identity.Test.Core.UIAutomation;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OpenQA.Selenium;
+
+namespace Microsoft.Identity.Test.Unit.Integration
+{
+    // Important: do not install a NuGet package with the Chrome driver as it is a security risk.
+    // Instead, install the Chrome driver on the test machine
+
+    // Note: these tests require permission to a KeyVault Microsoft account;
+    // Please ignore them if you are not a Microsoft FTE, they will run as part of the CI build
+    [TestClass]
+    [TestCategory(TestCategories.Selenium)]
+    [TestCategory(TestCategories.LabAccess)]
+    public class DeviceCodeFlow
+    {
+        private static readonly string[] Scopes = { "User.Read" };
+        private SeleniumWrapper _seleniumWrapper;
+
+        #region MSTest Hooks
+        /// <summary>
+        /// Initialized by MSTest (do not make private or readonly)
+        /// </summary>
+        public TestContext TestContext { get; set; }
+
+        [ClassInitialize]
+        public static void ClassInitialize(TestContext context)
+        {
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+        }
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            TestCommon.ResetState();
+            _seleniumWrapper = new SeleniumWrapper();
+        }
+
+        [TestCleanup]
+        public void Cleanup()
+        {
+            _seleniumWrapper?.Dispose();
+        }
+        #endregion
+
+        [TestMethod]
+        [Timeout(1 * 60 * 1000)] // 1 min timeout
+        public async Task DeviceCodeFlowTestAsync()
+        {            
+            LabResponse labResponse = LabUserHelper.GetDefaultUser();
+
+            Trace.WriteLine("Calling AcquireTokenWithDeviceCodeAsync");
+            PublicClientApplication pca = new PublicClientApplication(labResponse.AppId);
+            var result = await pca.AcquireTokenWithDeviceCodeAsync(Scopes, deviceCodeResult =>
+            {
+                RunAutomatedDeviceCodeFlow(deviceCodeResult, labResponse.User);
+
+                return Task.FromResult(0);
+            }).ConfigureAwait(false);
+
+            Trace.WriteLine("Running asserts");
+
+            Assert.IsNotNull(result);
+            Assert.IsTrue(!string.IsNullOrEmpty(result.AccessToken));
+        }
+
+        private void RunAutomatedDeviceCodeFlow(DeviceCodeResult deviceCodeResult, LabUser user)
+        {
+            IWebDriver driver = _seleniumWrapper.Driver;
+            try
+            {
+                Trace.WriteLine("Browser is open. Navigating to the Device Code url and entering the code");
+
+                driver.Navigate().GoToUrl(deviceCodeResult.VerificationUrl);
+                driver.FindElement(By.Id("code")).SendKeys(deviceCodeResult.UserCode);
+
+                IWebElement continueBtn = _seleniumWrapper.WaitForElementToBeVisibleAndEnabled(By.Id("continueBtn"));
```

Comment:
nit: consistency of having the ID values in a constants file or inline.  some ("continueBtn") are inline adn others (fields.SignInbuttonId) are in constants...